src/act.ts

```diff
@@ -359,18 +321,113 @@ export class Act {
             problemMatchers: [],
             runOptions: {},
             group: TaskGroup.Build,
-            execution: new ShellExecution(
-                command,
-                {
-                    cwd: commandArgs.path,
-                    env: settings.secrets
-                        .filter(secret => secret.value)
-                        .reduce((previousValue, currentValue) => {
-                            previousValue[currentValue.key] = currentValue.value;
-                            return previousValue;
-                        }, {} as Record<string, string>)
+            execution: new CustomExecution(async (resolvedDefinition: TaskDefinition): Promise<Pseudoterminal> => {
+                // Add new entry to workspace history
+                this.historyManager.workspaceHistory[commandArgs.path].push({
+                    index: historyIndex,
+                    count: count,
+                    name: `${commandArgs.name}`,
+                    status: HistoryStatus.Running,
+                    date: {
+                        start: start.toString()
+                    },
+                    taskExecution: taskExecution,
+                    commandArgs: commandArgs,
+                    logPath: logPath
+                });
+                historyTreeDataProvider.refresh();
+                this.storageManager.update(StorageKey.WorkspaceHistory, this.historyManager.workspaceHistory);
+
+                const writeEmitter = new EventEmitter<string>();
+                const closeEmitter = new EventEmitter<number>();
+
+                writeEmitter.event(async data => {
+                    try {
+                        // Create log file if it does not exist
+                        try {
+                            await fs.access(logPath);
+                        } catch (error: any) {
+                            await fs.writeFile(logPath, '');
+                        }
+
+                        // Append data to log file
+                        await fs.appendFile(logPath, data);
+                    } catch (error) { }
+                });
+
+                const handleIO = (data: any) => {
+                    const lines: string[] = data.toString().split('\n').filter((line: string) => line != '');
+                    for (const line of lines) {
+                        writeEmitter.fire(`${line.trimEnd()}\r\n`);
+                    }
                 }
-            )
+
+                const exec = childProcess.spawn(
+                    command,
+                    {
+                        cwd: commandArgs.path,
+                        shell: true,
```

Comment:
Note: This was added due to https://github.com/nodejs/node/issues/7367#issuecomment-229721296